book/src/using-bitfields.md

```diff
@@ -0,0 +1,196 @@
+```
+
+will print out
+
+```text
+a set to 1
+b set to 1
+c set to 3
+bitfield: a:1, b:1, c:3
+```
+
+Overflowing a bitfield will result in the same behavior as in C/C++: the bitfield will be set to 0.
+
+```rust,ignore
+    let mut bfield = unsafe { create_bitfield() };
+    bfield.set_a(1);
+    bfield.set_b(1);
+    bfield.set_c(12);
+    println!("c set to {} due to overflow", bfield.c());
+    unsafe { print_bitfield(bfield) };
+```
+
+will print out
+
+```text
+c set to 0 due to overflow
+bitfield: a:1, b:1, c:0
+```
+
+To create a new bitfield in Rust, use mem::zeroed.
+
+```rust,ignore
+    let mut bfield = unsafe { mem::zeroed::<bitfield>() };
+    bfield.set_a(1);
+    bfield.set_c(2);
+    unsafe { print_bitfield(bfield) };
+```
+
+This will print out
+
+```text
+bitfield: a:1, b:0, c:2
+```
+
+## Unsigned bitfield gotcha
+
+Special care should be given to handling unsigned bitfields. Let's change 'c' field to an int. C's valid values are now -2, -1, 0, and 1.
+
+```c
+typedef struct {
+    unsigned int a: 1;
+    unsigned int b: 1;
+    int c: 2;
+
+} bitfield;
+```
+
+Let's now walk the bitfield possibilities in both C and Rust up to the first overflow on each end:
+
+```c
+void walk_bitfield() {
+    bitfield bfield = create_bitfield();
+    print_bitfield(bfield);
+    bfield.c = 1;
+    print_bitfield(bfield);
+    bfield.c = 2;
+    print_bitfield(bfield);
+    bfield.c = 3;
+    print_bitfield(bfield);
+    bfield.c = 4;
+    print_bitfield(bfield);
+    bfield.c = -1;
+    print_bitfield(bfield);
+    bfield.c = -2;
+    print_bitfield(bfield);
+    bfield.c = -3;
+    print_bitfield(bfield);
+    bfield.c = -4;
+    print_bitfield(bfield);
+}
+```
+
+```rust,ignore
+fn walk_bitfield() {
+    let mut bfield = unsafe { mem::zeroed::<bitfield>() };
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(1);
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(2);
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(3);
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(4);
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(-1);
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(-2);
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(-3);
+    unsafe { print_bitfield(bfield) };
+    bfield.set_c(-4);
+    unsafe { print_bitfield(bfield) };
+}
+```
+
+This produces identical results in C's print function:
+
+```text
+Print C walk
+bitfield: a:0, b:0, c:0
+bitfield: a:0, b:0, c:1
+bitfield: a:0, b:0, c:-2
+bitfield: a:0, b:0, c:-1
+bitfield: a:0, b:0, c:0
+bitfield: a:0, b:0, c:-1
+bitfield: a:0, b:0, c:-2
+bitfield: a:0, b:0, c:1
+bitfield: a:0, b:0, c:0
+Print Rust walk
+bitfield: a:0, b:0, c:0
+bitfield: a:0, b:0, c:1
+bitfield: a:0, b:0, c:-2
+bitfield: a:0, b:0, c:-1
+bitfield: a:0, b:0, c:0
+bitfield: a:0, b:0, c:-1
+bitfield: a:0, b:0, c:-2
+bitfield: a:0, b:0, c:1
+bitfield: a:0, b:0, c:0
+```
+
+However, Bindgen's getter provides different results: 
```

Comment:
This is a bug, that we should fix, not behavior that we should document and canonicalize.

Let's remove this section and file a new issue to fix this bug instead.